Version bump and CHANGELOG update for 7.1 (#1563)
ROCm/rocm-systems: projects/rocprofiler-sdk/CHANGELOG.md

### Removed

- Support for compilation of gfx940 and gfx941 targets.

## ROCprofiler-SDK 1.1.0 for ROCm release 7.1

### Added

> - Process attach/detach - rocprofiler-sdk and rocprofv3 now allows dynamic profiling of already running GPU applications by attaching to their process ID (PID), rather than launching them through the profiler itself.

Reviewer: ```suggestion
- Dynamic process attachment: ROCprofiler-SDK and `rocprofv3` now facilitate dynamic profiling of a running GPU application by attaching to its process ID (PID), rather than launching the application through the profiler itself.
```

> - Scratch-memory trace information to the Perfetto output in rocprofV3.

Reviewer: ```suggestion
- Scratch-memory trace information to the Perfetto output in `rocprofv3`.
```

> - New capabilities to ATT support in rocprofV3, including realtime clock support for thread trace alignment on GFX9, enabling high-resolution clock computation and better synchronization across shader engines. Additionally,MultiKernelDispatch ATT support is now available across all ASICs.

Reviewer: ```suggestion
- New capabilities to the thread trace support in `rocprofv3`, including real-time clock support for thread trace alignment on gfx9 architecture. This enables high-resolution clock computation and better synchronization across shader engines. Additionally, `MultiKernelDispatch` thread trace support is now available across all ASICs.
```

> - Documentation for process attach/detach.

Reviewer: ```suggestion
- Documentation for dynamic process attachment.
```

> - Documentation for rocpd summaries.

Reviewer: ```suggestion
- Documentation for `rocpd` summaries.
```

> ### Changed

Reviewer: ```suggestion
### Optimized
```

> - Improved the stability and robustness of the RocPD output.

Reviewer: ```suggestion
- Improved the stability and robustness of the `rocpd` output.
```
